indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -654,6 +677,19 @@ private void announceTask(
     log.info("Worker[%s] reportin' for duty!", worker.getHost());
 
     try {
+      ScheduledFuture previousCleanup = removedWorkerCleanups.remove(worker.getHost());
+      if (previousCleanup != null) {
+        log.info("cancelling scheduled task cleanup for Worker[%s]", worker.getHost());
+        if (!previousCleanup.cancel(false)) {
+          try {
+            previousCleanup.get();
```

Comment:
is there a reason to even attempt to call `get()`? If we are unable to cancel the future the tasks will be cleaned up regardless